File: Cpp/869_Reordered_Power_of_2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int N) {
        string N_str = to_string(N);
        sort(N_str.begin(), N_str.end());
        for(int i=0; i<31; ++i) {
            string str = to_string(1<<i);
            if(N_str.size() != str.size()) continue;
            else if(N_str.size() < str.size()) return false;
            
            sort(str.begin(), str.end());
            if(N_str == str) return true;
        }
        return false;
    }
};

File: Kotlin/869_Reordered_Power_of_2.kt
class Solution {
    fun reorderedPowerOf2(N: Int): Boolean {
        val s = N.toString().sort()
        
        for(i in 0..30){
            val power_of_2 = (1 shl i).toString()
            if(s.length != power_of_2.length) continue
            else if(s.length < power_of_2.length) return false
            
            if(s == power_of_2.sort()) return true
        }
        
        return false
    }
}

fun String.sort() = String(toCharArray().apply{ sort() })